internal/appconfig/service.go

```diff
@@ -44,6 +45,14 @@ type ServiceHTTPCheck struct {
 	HTTPHeaders       map[string]string `json:"headers,omitempty" toml:"headers,omitempty"`
 }
 
+type ServiceMachineCheck struct {
+	Command     []string      `json:"command,omitempty" toml:"command,omitempty"`
+	Image       string        `json:"image,omitempty" toml:"image,omitempty"`
+	Entrypoint  []string      `json:"entrypoint,omitempty" toml:"entrypoint,omitempty"`
+	KillSignal  string        `json:"kill_signal,omitempty" toml:"kill_signal,omitempty"`
```

Comment:
KillSignal is `*string` on top level, I would use the same here as it makes propagating the value much easier.